lib/modules/hosts/eroshare.js

```diff
@@ -6,14 +6,40 @@ export default new Host('eroshare', {
 	name: 'eroshare',
 	domains: ['eroshare.com'],
 	logo: '//eroshare.com/favicon.ico',
-	detect: ({ pathname }) => (/^\/((?:i\/)?[a-z0-9]{8})/i).exec(pathname),
-	handleLink(href, [, hash]) {
-		return {
-			type: 'IFRAME',
-			embed: `//eroshare.com/embed/${hash}`,
-			width: '550px',
-			height: '550px',
-			fixedRatio: true,
-		};
+	detect({ pathname, href }) {
+		const slugRegex = /^\/((?:i\/)?[a-z0-9]{8})/i;
+		const slugDirectLinkRegex = /^https?:\/\/i\.eroshare\.com\/([a-z0-9_]{8})([a-zA-Z_]+)?\.[a-z0-9]{1,4}/i;
+		const slugMatch = slugRegex.exec(pathname);
+		const slugDirectLinkMatch = slugDirectLinkRegex.exec(href);
+
+		if(slugDirectLinkMatch){
```

Comment:
How about returning `[ 'album', slu...tch[1]]` or ` ['image,  href.url] `, and moving the `handle...` logic back inside the `handleLink` function?